pkg/cri/cri.go

```diff
@@ -0,0 +1,79 @@
+package cri
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+
+	log "github.com/cihub/seelog"
+)
+
+const (
+	criSocketPath = "unix:///var/run/cri.sock"
+)
+
+// ContainerInfo provides container information
+type ContainerInfo struct {
+	ID        string
+	Namespace string
+	Name      string
+	K8SUID    string
+}
+
+type APIs interface {
+	GetRunningContainers() (map[string]*ContainerInfo, error)
+}
+
+type Client struct{}
+
+func New() *Client {
+	return &Client{}
+}
+
+func (c *Client) GetRunningContainers() (map[string]*ContainerInfo, error) {
+	conn, err := grpc.Dial(criSocketPath, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	client := runtimeapi.NewRuntimeServiceClient(conn)
+
+	// List all ready sandboxes from the CRI
+	filter := &runtimeapi.PodSandboxFilter{
+		State: &runtimeapi.PodSandboxStateValue{
+			State: runtimeapi.PodSandboxState_SANDBOX_READY,
+		},
+	}
+	sandboxes, err := client.ListPodSandbox(context.Background(), &runtimeapi.ListPodSandboxRequest{Filter: filter})
+	if err != nil {
+		return nil, err
+	}
+
+	containerInfos := make(map[string]*ContainerInfo)
+	for _, sandbox := range sandboxes.Items {
+		if sandbox.Metadata == nil {
+			continue
+		}
+		uid := sandbox.Metadata.Uid
+
+		// Verify each pod only has one active sandbox. Kubelet will clean this
+		// up if it happens, so we should abort and wait until it does.
+		other, ok := containerInfos[uid]
+		if ok {
```

Comment:
nit... you could combine 64-65 on a single line:

```go
if other, ok := containerInfos[uid]; ok {
```

Reply:
Yup, you're right! (A previous version had the conditional the other way around and that didn't work.) Fixed.